Пул реквест на ревью

mdprof opened the pull request:
Пул реквест по первой задаче яндекс-практикума на ревью

gusar-off commented:
У тебя уже был какой-то код в ветке `main`, поэтому в данном пулл реквесте есть конфликты и лишние изменения.
Тебе нужно было создать пустой проект, залить его в основную ветку, далее сделать ветку для выполнения задания от основной ветки. Залить в созданную ветку весь код в рамках задания и сделать пулл руквест в основную ветку.

mdprof committed:
second commit

gusar-off commented on src/main/java/Main.java:
```diff
@@ -1,9 +1,101 @@
+    //Метод getNumberOfPeople() запрашивает ввод с консоли значения и помещает его в calculator.numberOfPeople,
+    // если значение меньше 1 или не является целым числом, то запрашивает повторно.
+    public static void getNumberOfPeople(Calculator calculator) {
+        Scanner sc = new Scanner(System.in);
```
Сканнер лучше создать один раз и передавать в методы параметром.

mdprof replied:
исправлено

gusar-off commented on src/main/java/Main.java:
```diff
@@ -1,9 +1,101 @@
+    public static String wordRoubleEnding(double price) {
+        int roundedPrice = (int) Math.floor(price);
+        String roubleEnding;
+        if (roundedPrice % 100 >= 11 && roundedPrice % 100 <= 19) {
```
на самом деле верхняя граница тут - 14, потому что остальные окончания - 15-16-17-18-19 будут такие же как у 5-6-7-8-9

mdprof replied:
исправлено, вроде очевидно, а я не догадался...

gusar-off commented on src/main/java/Main.java:
```diff
@@ -1,9 +1,101 @@
-        // вы не должны ограничиваться только классом Main и можете создавать свои классы по необходимости
-        System.out.println("Привет Мир");
 
+        Calculator calculator = new Calculator("", 0);
```
Можно не передавать это через конструктор, а инициализировать внутри или сразу при инициализации полей

mdprof replied:
Исправлено. Теперь начальные значения параметров присваиваются на уровне класса при инициализации.

mdprof committed:
После правок по ревью.

1.
Замечание:
>>public static void getNumberOfPeople(Calculator calculator) {
>>        Scanner sc = new Scanner(System.in);
>>Сканнер лучше создать один раз и передавать в методы параметром.

Решение:
Scanner инициализируется один раз и передается в методы как параметр.

2.
Замечание:
>>Calculator calculator = new Calculator("", 0);
>>Можно не передавать это через конструктор, а инициализировать внутри или сразу при инициализации полей

Решение:
Начальные значения полей Calculator теперь присваиваются сразу в описании класса.

3.
Замечание:
>>if (roundedPrice % 100 >= 11 && roundedPrice % 100 <= 19)
>>на самом деле верхняя граница тут - 14, потому что остальные окончания - 15-16-17-18-19 будут такие же как у 5-6-7-8-9

Решение:
вместо if (roundedPrice % 100 >= 11 && roundedPrice % 100 <= 19)
теперь if (roundedPrice % 100 >= 11 && roundedPrice % 100 <= 14)

mdprof committed:
После правок по ревью.

1.
Замечание:
>>public static void getNumberOfPeople(Calculator calculator) {
>>        Scanner sc = new Scanner(System.in);
>>Сканнер лучше создать один раз и передавать в методы параметром.

Решение:
Scanner инициализируется один раз и передается в методы как параметр.

2.
Замечание:
>>Calculator calculator = new Calculator("", 0);
>>Можно не передавать это через конструктор, а инициализировать внутри или сразу при инициализации полей

Решение:
Начальные значения полей Calculator теперь присваиваются сразу в описании класса.

3.
Замечание:
>>if (roundedPrice % 100 >= 11 && roundedPrice % 100 <= 19)
>>на самом деле верхняя граница тут - 14, потому что остальные окончания - 15-16-17-18-19 будут такие же как у 5-6-7-8-9

Решение:
вместо if (roundedPrice % 100 >= 11 && roundedPrice % 100 <= 19)
теперь if (roundedPrice % 100 >= 11 && roundedPrice % 100 <= 14)
